Makefile

```diff
@@ -0,0 +1,53 @@
+DEFAULT_GOAL := help
+
+# Include .env and export it so variables set in there are available in the
+# Makefile.
+include .env
+export
+
+# Set these in the environment to override them. This is helpful for
+# development if you have file ownership problems because the user in the
+# container doesn't match the user on your host.
+USER_ID ?= 10001
+GROUP_ID ?= 10001
+
+.PHONY: help
+help:
+	$(info Available rules:)
+	$(info )
+	@fgrep -h "##" Makefile | fgrep -v fgrep | sed 's/\(.*\):.*##/\1:/' | column -t -s '|'
+
+.docker-build:
+	make build
+
+.env:
+	@if [ ! -f .env ]; \
+	then \
+	echo "Creating .env file..."; \
+	echo "# USER_ID=\n# GROUP_ID=\n" > .env; \
+	fi
+
+.PHONY: build
+build: .env  ## | Build docker image
+	docker-compose build --build-arg USER_ID=${USER_ID} --build-arg GROUP_ID=${GROUP_ID} app
+	touch .docker-build
```

Comment:
The `.docker-build` is a real file, so then it'll get handled by make and targets that depend on it will build the image if it needs building.